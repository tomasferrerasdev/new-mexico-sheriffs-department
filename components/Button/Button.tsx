import { css, SerializedStyles } from '@emotion/react';
import styled from '@emotion/styled';
import { MouseEvent } from 'react';

export type Color = 'primary' | 'disabled';

export type Props = {
  children: string;
  color?: Color;
  onClick: (event: MouseEvent<HTMLButtonElement>) => void;
};

export const getColor = (color?: Color): SerializedStyles => {
  switch (color) {
    case 'primary':
      return css`
        background-color: #fdbc0f;
        color: #172634;
      `;
    case 'disabled':
      return css`
        background-color: #c6c6c6;
        color: #475869;
      `;
    default:
      return css``;
  }
};

export const Button = styled.button<Props>`
  all: unset;
  display: flex;
  justify-self: center;
  align-items: center;
  justify-content: center;
  user-select: none;
  cursor: pointer;
  font-size: 1rem;
  ${({ color }) => getColor(color)}
  border: 1px solid #172634;
  padding: 0.75rem 1rem;
  &:hover {
    opacity: 0.9;
  }
`;

Button.defaultProps = {
  color: 'primary',
};
