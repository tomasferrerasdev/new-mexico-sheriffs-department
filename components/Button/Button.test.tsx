import { render } from '@testing-library/react';
import { Button } from './Button';

describe('Button test cases', () => {
  it('Render check', () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <Button onClick={onClick}>REGISTER NOW</Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  /*
  it('Check onClick callback', () => {
    const onClick = jest.fn();

    render(<Button onClick={onClick}>REGISTER NOW</Button>);

    const element = screen.getByRole('button');
    userEvent.click(element);
    expect(onClick).toHaveBeenCalled();
  });
  */
});
