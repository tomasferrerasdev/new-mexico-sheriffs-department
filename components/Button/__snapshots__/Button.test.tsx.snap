// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases Render check 1`] = `
<DocumentFragment>
  <button
    class="css-1bdixvh"
    color="primary"
  >
    REGISTER NOW
  </button>
</DocumentFragment>
`;
